cmd/kubedownscaler/main.go

```diff
@@ -80,15 +83,64 @@ func main() {
 		os.Exit(1)
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
+
+	defer cancel()
+
+	downscalerNamespace, err := kubernetes.GetCurrentNamespace()
+	if err != nil {
+		slog.Warn("couldn't get namespace or running outside of cluster; skipping leader election", "error", err)
+		startScanning(client, ctx, &layerCli, &layerEnv, config)
+
+		return
+	}
+
+	lease, err := client.CreateLease(leaseName, downscalerNamespace)
+	if err != nil {
+		slog.Warn("failed to create lease", "error", err)
+		slog.Warn("proceeding without leader election, this may cause multiple instances to conflict when modifying the same resources")
+		startScanning(client, ctx, &layerCli, &layerEnv, config)
+	}
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		<-sigs
+		cancel()
+	}()
+
+	leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
+		Lock:            lease,
+		ReleaseOnCancel: true,
+		LeaseDuration:   60 * time.Second,
+		RenewDeadline:   15 * time.Second,
+		RetryPeriod:     5 * time.Second,
```

Comment:
Also, on the topic of the TODO for fine tuning these values. 
I think it would be smart to something more like this:
```suggestion
		LeaseDuration:   30 * time.Second,
		RenewDeadline:   20 * time.Second,
		RetryPeriod:     5 * time.Second,
```
Its common to have the RenewDeadline be 2/3 of the LeaseDuration. I think the RetryPeriod being 5s is fine, since its not like the downscaler needs to be super high availability and we already send a lot of api requests as it is. I think the Lease Duration being 30s is a good middle ground for the downscaler, it means that the leader still has 3-4 chances at renewing and also 30s being the default Scan interval this means that at least when using the default interval there is basically 0 downtime between scans. I would go against retrieving these values from the specified scan interval, because if these values are too low it will cause a lot more leader switches and in turn unnecessary overhead.